assets/src/design-system/components/input/useNumericInput.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import Big from 'big.js';
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
+/**
+ * Internal dependencies
+ */
+import { isInputValid, parseInput } from './validation';
+
+export const useNumericInput = ({
+  allowEmpty,
+  isFloat,
+  max,
+  min,
+  onChange,
+  value,
+}) => {
+  const inputRef = useRef(null);
+  const oldValue = useRef(value);
+  const revertToOriginal = useRef(false);
+  const [currentValue, setCurrentValue] = useState(value);
+  const options = useMemo(() => ({ allowEmpty, isFloat, max, min }), [
+    allowEmpty,
+    isFloat,
+    max,
+    min,
+  ]);
+
+  /**
+   * Call external `onChange`
+   */
+  const handleBlur = useCallback(
+    (ev) => {
+      let newValue = parseInput(oldValue.current, options);
+
+      if (!revertToOriginal.current) {
+        newValue = parseInput(currentValue, options);
+      }
+
+      revertToOriginal.current = false;
+
+      setCurrentValue(newValue);
+      onChange(ev, newValue);
+    },
+    [currentValue, onChange, options]
+  );
```

Comment:
The only issue I can find is here - if you type something illegal and blur or press enter (both are routed through this callback), the input is left as "null", where it instead should revert to the original text ("last known valid state"). Otherwise, you can get his confusing output of `nullms`:

![invalid-input-cut](https://user-images.githubusercontent.com/637548/108427246-ddb81400-720a-11eb-8542-58e38d93c9b0.gif)

Reply:
Sorry about the force push - I rebased `main`.

The new commit is this: https://github.com/google/web-stories-wp/pull/6419/commits/5a2d108f0e436caf98009d17be1f56dccb73fe8c